nameko/containers.py

```diff
@@ -348,11 +364,15 @@ def _run_worker(self, worker_ctx, handle_result):
 
         with _log_time('ran worker %s', worker_ctx):
 
-            self.dependencies.injections.all.inject(worker_ctx)
+            for dependency in self.dependencies:
```

Comment:
Yeah, there's potentially quite a lot of I/O going on in ``acquire_injection``, depending on how lazy the injections are.